src/Struct/Exprs.hs

```diff
@@ -14,16 +16,16 @@ data UsProgs = UsProgs [UsProg] UsTm    -- definitions, main
 
 -- Individual user-level definition
 data UsProg =
-    UsProgDefine Var UsTm Type             -- lhs, rhs, type
-  | UsProgExtern Var Type               -- lhs, type
-  | UsProgData Var [Var] [Ctor]         -- lhs, type params, constructors
+    UsProgDefine TmName UsTm Type       -- lhs, rhs, type
+  | UsProgExtern TmName Type            -- lhs, type
+  | UsProgData TpName [TpVar] [Ctor]    -- lhs, type params, constructors
   deriving (Eq, Ord)
 
 -- Scheme-ified definition
 data SProg =
-    SProgDefine Var [Var] [Var] Term Type  -- lhs, tags, type params, rhs, type
-  | SProgExtern Var Type                -- lhs, type
-  | SProgData Var [Var] [Var] [Ctor]    -- lhs, tags, type params, constructors
+    SProgDefine TmName [Tag] [TpVar] Term Type  -- lhs, tags, type params, rhs, type
```

Comment:
You decided not to separate tags from tag variables?

Reply:
Yeah, we can always split them later if needed...